### PR TITLE
Update get probe code to make it work it IOS XE 16.12.x

### PR DESCRIPTION
**Pull request by javcasalc**
This patch makes ios.py get_probes_config() getter able to work with new IOS XE versions:
```
csr1000v#show version | i , Version 
Cisco IOS XE Software, Version 16.12.03
Cisco IOS Software [Gibraltar], Virtual XE Software (X86_64_LINUX_IOSD-UNIVERSALK9-M), Version 16.12.3, RELEASE SOFTWARE (fc5)
```
```
csr1000v#sh run | section ip sla     
ip sla 1
 icmp-echo 10.0.0.2 source-interface GigabitEthernet11
 tag napalm_test
 frequency 15
 history buckets-kept 60
ip sla schedule 1 life forever start-time now
csr1000v#
```

This code is pretty inflexible as it is based on python regex multiline, so the block (ip sla config) must match all lines and options. It should be improved.

### CODE REVIEW

**Comment by javcasalc** [comment]
https://github.com/napalm-automation/napalm/issues/1281

File: napalm/ios/ios.py [hunk 1]
```diff
@@ -2697,18 +2697,22 @@ def process_mac_fields(vlan, mac, mac_type, interface):
...
+            r"\s+buckets-kept\s+(?P<probe_count0>\d+))|(\s+history\s+buckets-kept"
+            r"\s+(?P<probe_count1>\d+)\n.*\s+frequency\s+(?P<interval1>\d+)))"
         )
+
         probe_args = {
             "icmp-echo": r"^(?P<target>\S+)\s+source-(?:ip|interface)\s+(?P<source>\S+)$"
         }
         probe_type_map = {"icmp-echo": "icmp-ping"}
-        command = "show run | include ip sla [0-9]"
+        command = "show run | section ip sla [0-9]"
```
Comment (mirceaulinic) [inline napalm/ios/ios.py:2715]: I don't have a lot of experience on IOS platforms, but my understanding from past PRs / conversations is that older versions may not support `| section` (while `| include` should always be there). Could you clarify what's the advantage or why you chose to switch @javcasalc?

**Review by mirceaulinic — CHANGES_REQUESTED** [review]
Thanks for this PR @javcasalc. The regex looks good, but in make it fruitful, we'll need a new test case with the output you're seeing on your platform (to show why the current code is not working, and prove that the new regex would solve it).

File: napalm/ios/ios.py (hunk 1 above)
Comment (javcasalc) [reply napalm/ios/ios.py:2715]: According to Cisco doc, the right flag is ```section```: https://www.cisco.com/c/en/us/td/docs/ios/fundamentals/command/reference/cf_book/cf_s1.html

```
Examples

The following examples compare the filtering characteristics of the show running-config | include command with the show running-config | section command. The first example gathers just the lines from the configuration file with "interface" in them.

Router# show running-config | include interface


interface Ethernet0/0 
interface Ethernet1/0 
interface Serial2/0 
interface Serial3/0


The next example uses the show command section command to gather the lines in the configuration file with "interface" in them as well as any lines associated with those entries. In this example, interface configuration information is captured.

Router# show running-config | section include interface


interface Ethernet0/0 
 shutdown 
 no cdp enable

interface Ethernet1/0 
 shutdown 
 no cdp enable 
interface Serial2/0 
 shutdown 
 no cdp enable 
interface Serial3/0 
 shutdown 
 no cdp enable
```

I did some tests in GNS3 with some old cisco routers (cisco 3640 with IOS 12.4; cisco 7200 with IOS 15.2) and I do confirm that ```section``` is the right choice:
imagen


Regarding the test case, I've included an extra "ip sla 3" entry, formatted with the new ios format, in order to test the correct parsing of both styles (old and new styles). Is this ok?

Comment (mirceaulinic) [reply napalm/ios/ios.py:2715]: Fair enough... any thoughts on this @ktbyers ?

**Review by mirceaulinic — APPROVED** [review]

**Pull request merged**